src/main/java/org/betonquest/betonquest/compatibility/protocollib/NPCGlow.java

```diff
@@ -0,0 +1,315 @@
+import net.citizensnpcs.api.event.NPCSpawnEvent;
+import net.citizensnpcs.api.npc.NPC;
+import org.apache.commons.lang3.EnumUtils;
+import org.betonquest.betonquest.BetonQuest;
+import org.betonquest.betonquest.api.config.quest.QuestPackage;
+import org.betonquest.betonquest.compatibility.protocollib.api.GlowAPI;
+import org.betonquest.betonquest.config.Config;
+import org.betonquest.betonquest.exceptions.ObjectNotFoundException;
+import org.betonquest.betonquest.id.ConditionID;
+import org.betonquest.betonquest.utils.PlayerConverter;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.configuration.ConfigurationSection;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerQuitEvent;
+import org.bukkit.scheduler.BukkitRunnable;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * NPCGlow class
+ */
+@SuppressWarnings({"PMD.TooManyMethods", "PMD.GodClass"})
+@CustomLog
+public final class NPCGlow extends BukkitRunnable implements Listener {
+
+    /**
+     * instance of GlowAPI
+     */
+    private GlowAPI glowAPI;
+
+    /**
+     * instance of NPCGlow
+     */
+    private static NPCGlow instance;
+    /**
+     * List of conditions for specific NPC that will get glow
+     */
+    private final Map<Integer, Set<ConditionID>> npcs;
+    /**
+     * List player that can seen glowing npc.
+     */
+    private final Map<Integer, Collection<Player>> npcPlayersMap;
+    /**
+     * Store NPCID for each color
+     */
+    private final Map<Integer, ChatColor> npcColor;
+
+    /**
+     * Constructor of NPCGlow instance
+     */
+    public NPCGlow() {
+        super();
+        npcPlayersMap = new HashMap<>();
+        npcs = new HashMap<>();
+        npcColor = new HashMap<>();
+        glowAPI = new GlowAPI();
+        Bukkit.getScheduler().runTaskLater(BetonQuest.getInstance(), this::loadFromConfig, 5L);
+        runTaskTimer(BetonQuest.getInstance(), 0, 5);
+        Bukkit.getPluginManager().registerEvents(this, BetonQuest.getInstance());
+    }
+
+    /**
+     * Starts (or restarts) the NPCGlow. It loads the current configuration for hidden NPCs
+     */
+    public static void start() {
+        synchronized (NPCGlow.class) {
+            if (instance != null) {
+                instance.stop();
+            }
+            instance = new NPCGlow();
+        }
+    }
+
+    /**
+     * @return the currently used NPCGlow instance
+     */
+    public static NPCGlow getInstance() {
+        return instance;
+    }
+
+    /**
+     * load all data from config
+     */
+    @SuppressWarnings({"PMD.ShortVariable", "PMD.NPathComplexity", "PMD.CognitiveComplexity", "PMD.CyclomaticComplexity"})
```

Comment:
You should not supress that many thing, and i will continue my review, when this is solved